plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -558,6 +567,96 @@ public void updatePartitionStatistics(Table table, String partitionName, Functio
         removedStatistics.forEach(column -> deletePartitionColumnStatistics(table.getDbName(), table.getTableName(), partitionName, column));
     }
 
+    @Override
+    public void updatePartitionStatistics(Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        if (updatePartitionStatisticsBatchEnabled) {
+            if (updates.size() <= updatePartitionStatisticsBatchSize) {
+                updatePartitionStatisticsBatch(table, updates);
+                return;
+            }
+
+            Map<String, Function<PartitionStatistics, PartitionStatistics>> batch = new HashMap<>(updatePartitionStatisticsBatchSize);
+            updates.forEach((partitionName, update) -> {
+                batch.put(partitionName, update);
+                if (batch.size() >= updatePartitionStatisticsBatchSize) {
+                    updatePartitionStatisticsBatch(table, batch);
+                    batch.clear();
+                }
+            });
+            if (!batch.isEmpty()) {
+                updatePartitionStatisticsBatch(table, batch);
+                batch.clear();
+            }
+        }
+        else {
+            updates.forEach((partitionName, update) -> updatePartitionStatistics(table, partitionName, update));
+        }
+    }
+
+    private void updatePartitionStatisticsBatch(Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        List<String> partitionNames = updates.keySet().stream().collect(toImmutableList());
+        List<Partition> partitions = getPartitionsByNames(table.getDbName(), table.getTableName(), partitionNames);
+        checkState(partitionNames.size() == partitions.size(),
```

Comment:
Consider providing more information in case that the partition has been dropped during the query:

https://github.com/trinodb/trino/blob/28002f0500aac964d5a6cc84183b654a9abae3b6/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java#L306-L311